src/qt3utils/applications/qt3ple/main.py

```diff
@@ -0,0 +1,395 @@
+parser.add_argument('-cmap', metavar='<MPL color>', default='gray',
+                    help='Set the MatplotLib colormap scale')
+
+args = parser.parse_args()
+
+logger = logging.getLogger(__name__)
+logging.basicConfig()
+
+if args.quiet is False:
+    logger.setLevel(logging.INFO)
+
+NIDAQ_DEVICE_NAMES = ['NIDAQ Rate Counter', "Lockin & Wavemeter"]
+RANDOM_DAQ_DEVICE_NAME = 'Random Data Generator'
+
+DEFAULT_DAQ_DEVICE_NAME = NIDAQ_DEVICE_NAMES[0]
+
+CONTROLLER_PATH = 'qt3utils.applications.controllers'
+STANDARD_CONTROLLERS = {NIDAQ_DEVICE_NAMES[0] : 'nidaq_rate_counter.yaml',
+                        NIDAQ_DEVICE_NAMES[1] : 'nidaq_wm_ple.yaml'}
+
+class ScanImage:
+    def __init__(self, mplcolormap='gray') -> None:
+        self.fig, self.ax = plt.subplots()
+        self.cbar = None
+        self.cmap = mplcolormap
+        self.cid = self.fig.canvas.mpl_connect('button_press_event', self.onclick)
+        self.ax.set_xlabel('Voltage')
+        self.log_data = False
+
+    def update(self, model) -> None:
+
+        if self.log_data:
+            data = np.log10(model.scanned_count_rate)
+            data[np.isinf(data)] = 0  # protect against +-inf
+        else:
+            data = model.scanned_count_rate
+        data = np.array(data).T.tolist()
+
+        self.artist = self.ax.imshow(data, cmap=self.cmap, extent=[model.current_t + model.raster_line_pause,
+                                                                   0,
+                                                                   model.vmin,
+                                                                   model.vmax + model.step_size
+                                                                   ])
+        if self.cbar is None:
+            self.cbar = self.fig.colorbar(self.artist, ax=self.ax)
+        else:
+            self.cbar.update_normal(self.artist)
+
+        if self.log_data is False:
+            self.cbar.formatter.set_powerlimits((0, 3))
+
+        self.ax.set_xlabel('Pixels')
+        self.ax.set_ylabel('Voltage (V)')
+
+    def reset(self) -> None:
+        self.ax.cla()
+
+    def set_onclick_callback(self, f) -> None:
+        self.onclick_callback = f
+
+    def onclick(self, event) -> None:
+        pass
+
+
+class SidePanel():
+    def __init__(self, root, scan_range) -> None:
+        frame = tk.Frame(root.root)
+        frame.pack(side=tk.LEFT, fill=tk.BOTH, expand=True)
+
+        row = 0
+        tk.Label(frame, text="Scan Settings", font='Helvetica 16').grid(row=row, column=0, pady=10)
+        row += 1
+        self.startButton = tk.Button(frame, text="Start Scan")
+        self.startButton.grid(row=row, column=0)
+        self.stopButton = tk.Button(frame, text="Stop Scan")
+        self.stopButton.grid(row=row, column=1)
+        self.saveScanButton = tk.Button(frame, text="Save Scan")
+        self.saveScanButton.grid(row=row, column=2)
```

Comment:
I would opt to refactor these definitions so that they follow the general python rule for attributes (e.g. `startButton -> start_button`).